src/sql/src/plan/statement/ddl/connection.rs

```diff
@@ -262,17 +262,22 @@ impl ConnectionOptionExtracted {
                     let mut duplicate_azs: BTreeSet<String> = BTreeSet::new();
                     // Validate each AZ is supported
                     for connection_az in &connection.availability_zones {
-                        if unique_azs.contains(connection_az) {
-                            duplicate_azs.insert(connection_az.to_string());
-                        } else {
-                            unique_azs.insert(connection_az.to_string());
-                        }
-                        if !supported_azs.contains(connection_az) {
+                        if !connection_az
+                            .chars()
+                            .all(|c| c.is_ascii_alphanumeric() || c == '-')
+                            || !connection_az.contains("-az")
```

Comment:
seems like we could tighten this up to be more precise. https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html#az-ids has more details on the structure

the most legit solution would be to automatically pull every AZ ID from the partition we're operating in during our Pulumi run and then plumbing it 20 layers deep to here... in practice I think just ensuring we're following the structure in the doc above is sufficient

Reply:
(and if we pull out some logic to validate the ids, we should add a quick unit test)